File: app/src/main/java/com/anymore/client/mvp/login/LoginContract.java
package com.anymore.client.mvp.login;

import com.anymore.mvpkit.mvp.base.BaseContract;
import io.reactivex.Observable;

/**
 * Created by liuyuanmao on 2019/7/16.
 */
public interface LoginContract {
    interface IView extends BaseContract.IBaseView{
        void loginSuccess();
    }

    interface IPresenter extends BaseContract.IBasePresenter{
        void mockLogin(String username,String password);
    }

    interface IModel extends BaseContract.IBaseModel{
        Observable<Boolean> login(String username,String password);
    }
}

File: app/src/main/java/com/anymore/client/ui/mvp/login/LoginContract.kt
package com.anymore.client.ui.mvp.login

import com.anymore.mvpkit.mvp.base.BaseContract
import io.reactivex.Observable

/**
 * Created by liuyuanmao on 2019/7/23.
 */
interface LoginContract {

    interface IView: BaseContract.IBaseView{
        fun loginSuccess()
    }

    interface IPresenter: BaseContract.IBasePresenter{
        fun login(username:String,password:String)
    }

    interface IModel: BaseContract.IBaseModel{
        fun login(username:String,password:String):Observable<Boolean>
    }
}